### DISCUSSION TITLE
Firebase = Function statements require a function name

### DISCUSSION THREAD

**Discussion by Schmell** [sveltejs/kit#8391] [Q&A]
When using firebase I get this error `Syntax Error: Function statements require a function name`  when I try to initialize firebase.
I am assuming it has something to do with **ssr** or **prerendering**, but I have another SvelteKit project that is working with firebase. I compared all of my config, +layout, +page files with this new project and I can't see that I turned off ssr or prerendering anywhere.
Also I see other peoples examples and they don't seem to indicate that I need some kind of config/setting to deal with this.

The code below throws an error
src/lib/firebase.ts
```ts
import { initializeApp } from 'firebase/app';
import { getFirestore } from 'firebase/firestore';
import { getAuth } from 'firebase/auth';

const firebaseConfig = {
   // redacted
};

// Initialize Firebase
export const app = initializeApp(firebaseConfig);
export const firestore = getFirestore(app);
export const auth = getAuth(app);
```

src/routes/+page.svelte
```svelte
<script lang="ts">
   import {firestore} from '$lib/firebase'
</script>

<h3>Hey from route/+page.svelte</h3>
```

I understand that we should use +page/+layout.ts for loading data,, but i am trying to use [sveltefire](https://github.com/codediodeio/sveltefire).
Also i just tried it and get the same error importing (initializing) in +layout.server.ts

**Comment by Bohdan-Dubravin** [comment]
Have the same issue when import getFirestore

**Reply by Schmell** [reply]
Problem is that firebase is **client only**. you need to 
`export const ssr = false;`
on pages that use it
